src/components/CSVButton/CSVButton.jsx

```diff
@@ -0,0 +1,258 @@
+        'Pre-Test A Notes',
+        'Pre-Test R',
+        'Pre-Test R Notes',
+      ];
+    }
+    if (type === 'area') {
+      return [
+        'Area ID',
+        'Student ID',
+        'First Name',
+        'Last Name',
+        'Grade',
+        'Home Teacher',
+        'Gender',
+        'Ethnicity',
+        'Student Group ID',
+        'Post-Test A',
+        'Post-Test A Notes',
+        'Post-Test R',
+        'Post-Test R Notes',
+        'Pre-Test A',
+        'Pre-Test A Notes',
+        'Pre-Test R',
+        'Pre-Test R Notes',
+      ];
+    }
+    if (type === 'site') {
+      return [
+        'Site ID',
+        'Student ID',
+        'First Name',
+        'Last Name',
+        'Grade',
+        'Home Teacher',
+        'Gender',
+        'Ethnicity',
+        'Student Group ID',
+        'Post-Test A',
+        'Post-Test A Notes',
+        'Post-Test R',
+        'Post-Test R Notes',
+        'Pre-Test A',
+        'Pre-Test A Notes',
+        'Pre-Test R',
+        'Pre-Test R Notes',
+      ];
+    }
+    return [];
+  }
+
+  function mapResponse() {
+    if (type === 'allAreas') {
+      return studentResponseData.map(student => [
+        student.areaId,
+        student.studentId,
+        student.firstName,
+        student.lastName,
+        student.grade,
+        student.homeTeacher,
+        student.gender,
+        student.ethnicity,
+        student.studentGroupId,
+        student.posttestA,
+        student.posttestANotes,
+        student.posttestR,
+        student.posttestRNotes,
+        student.pretestA,
+        student.pretestANotes,
+        student.pretestR,
+        student.pretestRNotes,
+      ]);
+    }
+    if (type === 'area') {
+      return studentResponseData.map(student => [
+        student.areaId,
+        student.studentId,
+        student.firstName,
+        student.lastName,
+        student.grade,
+        student.homeTeacher,
+        student.gender,
+        student.ethnicity,
+        student.studentGroupId,
+        student.posttestA,
+        student.posttestANotes,
+        student.posttestR,
+        student.posttestRNotes,
+        student.pretestA,
+        student.pretestANotes,
+        student.pretestR,
+        student.pretestRNotes,
+      ]);
+    }
+    if (type === 'site') {
+      return studentResponseData.map(student => [
+        student.siteId,
+        student.studentId,
+        student.firstName,
+        student.lastName,
+        student.grade,
+        student.homeTeacher,
+        student.gender,
+        student.ethnicity,
+        student.studentGroupId,
+        student.posttestA,
+        student.posttestANotes,
+        student.posttestR,
+        student.posttestRNotes,
+        student.pretestA,
+        student.pretestANotes,
+        student.pretestR,
+        student.pretestRNotes,
+      ]);
+    }
+    return [];
+  }
+
+  const CSVReport = {
+    data: studentResponseData,
+    headers: createHeaders(),
+    filename: 'Areas_Report.csv',
```

Comment:
can we make the file name different based on the button pressed?

Reply:
yes, will add